---
title: "Disp_Genetic_Analysis"
author: "Brandie QC"
date: "`r Sys.Date()`"
output: 
  html_document: 
    keep_md: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Analyzing the genetic data from the Close and Far dispersal trays

To Do:

-   Incorporate the Source populations

-   Try computing the diversity metrics and FST using sliding windows so that I can get estimates of variation across the genome

-   Compare ecotype frequencies (did some ecotypes disperse better?)

-   Figure out a way to test for bias in allele frequencies

-   Only compare FST within blocks

Next steps From KD:

1.  Compare genetic diversity in dispersed versus non-dispersed samples (source versus far, and near versus far)

2.  Determine whether the source/near versus far samples differ in allele freq at specific loci across the genome (Fst). If so, that would identify alleles that occur more in dispersed pops (bias). I think this would be a comparison between pairs to begin, rather than all possible trays. If you identify potential loci, then you could calculate the allele freq at that locus in all pops and compare between far and source using all pops, but there would be low power. If you DO find something, though that would be very convincing. There are probably more powerful ways to test for allele-freq diffs

3.  While far more challenging, it would be good to look at Moi's code to see whether you can identify ecotypes that have been dispersed, and compare the frequency of those ecotypes between source/near versus far trays--another metric of "bias". Same concerns as in b.

Block Info:

-   Block 1
    -   Sources: 2, 3, 6
    -   Close trays: 3, 6 (no data: 1, 4)
    -   Far tray: 7
-   Block 2
    -   Sources: 4, 7, 8
    -   Close trays: 9, 11 (no data: 12, 14)
    -   Far tray: 8
-   Block 3
    -   Sources: 11, 12, 15
    -   Close trays: 15, 20 (no data: 17, 18)
    -   Far tray: 21
-   Block 4
    -   Sources: 13, 16, 17

    -   Close trays: 23, 26 (no data: 25, 28)

    -   Far tray: 22

## Libraries

```{r}
library(tidyverse)

sem <- function(x, na.rm=FALSE) {           #for calculating standard error
  sd(x,na.rm=na.rm)/sqrt(length(na.omit(x)))
} 
```

## Prep Source Tray SNPs
```{r, eval=FALSE}
source_snps <- read_csv("../../Analyzing Genetic Data/GrENE-net Cages/site28_hapFIRE_allele_frequency.csv") %>%
  select(`28_1_2`:`28_1_4`, `28_1_6`:`28_1_8`, `28_1_11`:`28_1_13`, `28_1_15`:`28_1_17`) #only want gen 1 source trays (the flowers that dispersed the seeds to dispersal trays)
#also, only want source pops (not caged pops)

source_info <- read_csv("../../Analyzing Genetic Data/GrENE-net Cages/site28_sample_table.csv") %>%
  filter(generation==1) %>%  #only want gen 1 source trays (the flowers that dispersed the seeds to dispersal trays)
  filter(Treatment=="Open")

head(source_snps)
head(source_info)

## need to make "snp_frequency.txt" files for each source tray 
### need columns for: chrom	pos	SourceID.af
source_snps_IDs <- source_snps %>% 
  rename_with(~ str_remove(.x, "28_1_"), contains("28_1_")) %>% #remove identifiers for the site and generation 
  rename_with(~ paste0("S", .x, ".af")) #Add prefix of S for "source" trays and suffix for allele freq
names(source_snps_IDs)

#get chrom and pos info from one of the disp tray files (NOTE: I AM ASSUMING THE ROWS ARE IN THE SAME ORDER!!)
chrom_pos <- read_delim("../../Analyzing Genetic Data/DispersalTraps/hapFire_Results/SNPs_to_analyze/D03_snp_frequency.txt") %>% 
  select(chrom, pos)
head(chrom_pos)
dim(chrom_pos) #same number of rows as source trays 
dim(source_snps_IDs) 

source_snps_IDs_pos <- bind_cols(chrom_pos, source_snps_IDs)
head(source_snps_IDs_pos)
tail(source_snps_IDs_pos)
write_delim(source_snps_IDs_pos, "../../Analyzing Genetic Data/DispersalTraps/hapFire_Results/SNPs_to_analyze/SourcePops_snp_frequency.txt", delim = "\t")
```

## Load Sample Info

```{r}
sample_info <- read_csv("../../Analyzing Genetic Data/DispersalTraps/hapFire_Results/SNPs_to_analyze/DispSamples.csv")
head(sample_info)
```

## Load Genome-Wide Diversity

Nucleotide diversity (theta_pi) - This quantifies the average number of differences (mismatches) between any pair of DNA sequences in a sample The higher this statistic, the more differences you will expect to see between any pair of randomly sampled DNA sequences.

Watterson’s estimator (theta_w) - This quantifies the number of segregating sites within a sample, scaled by a constant. The higher this statistic, the more sites are found to be segregating (i.e., polymorphic).

Tajima’s D (tajima_d) - This quantifies the difference between Watterson’s estimator and nucleotide diversity, and scales the result by a constant. This statistic is useful because it focuses not on the magnitude but on the architecture of diversity in a sample, compared to a neutrally evolving population of constant size. Negative values imply an excess of rare alleles, and positive values imply a deficit of rare alleles, relative to a population at equilibrium.

```{r}
genome_wide_diversity <- read_csv("../../Analyzing Genetic Data/DispersalTraps/hapFire_Results/SNPs_to_analyze/diversity_genome.csv") 

genome_wide_diversity_diagnostics <- genome_wide_diversity %>% 
  select(total.masked:total.passed)
genome_wide_diversity_diagnostics #3227823 SNPs

genome_wide_diversity_short <- genome_wide_diversity %>% 
  select(D03.theta_pi:S17.tajimas_d) %>% 
  select(-contains("missing"), -contains("passed"), -contains("numeric"))

genome_wide_thetapi <- genome_wide_diversity_short %>% 
  select(contains("theta_pi")) %>% 
  pivot_longer(
    cols = ends_with("theta_pi"),
    names_to = "PopID",
    values_to = "Theta.Pi"
  ) %>% 
  mutate(PopID=str_remove(PopID, ".theta_pi"))

genome_wide_thetawatterson <- genome_wide_diversity_short %>% 
  select(contains("theta_watterson")) %>% 
  pivot_longer(
    cols = ends_with("theta_watterson"),
    names_to = "PopID",
    values_to = "Theta.Watterson"
  ) %>% 
  mutate(PopID=str_remove(PopID, ".theta_watterson"))

genome_wide_tajimasd <- genome_wide_diversity_short %>% 
  select(contains("tajimas_d")) %>% 
  pivot_longer(
    cols = ends_with("tajimas_d"),
    names_to = "PopID",
    values_to = "Tajimas.D"
  ) %>% 
  mutate(PopID=str_remove(PopID, ".tajimas_d"))

genome_wide_diversity_long <- sample_info %>% 
  left_join(genome_wide_thetapi) %>% 
  left_join(genome_wide_thetawatterson) %>% 
  left_join(genome_wide_tajimasd)
```

### Plots with all dispersal trays

```{r}
genome_wide_diversity_long %>% 
  ggplot(aes(x=fct_reorder(PopID, Theta.Pi), y=Theta.Pi, group = Treatment, fill=Treatment)) +
  geom_col(width=0.7, position=position_dodge(0.8)) + 
  theme_classic() + labs(x="Tray", fill="Distance") +
  scale_y_continuous(expand = c(0, 0)) +
  scale_fill_manual(values=c("#298B8A", "#B4A7E1", "#E69F00")) 

genome_wide_diversity_long %>% 
  ggplot(aes(x=fct_reorder(PopID, Theta.Watterson), y=Theta.Watterson, group = Treatment, fill=Treatment)) +
  geom_col(width=0.7, position=position_dodge(0.8)) + 
  theme_classic() + labs(x="Dispersal Tray", fill="Distance") +
  scale_y_continuous(expand = c(0, 0)) +
  scale_fill_manual(values=c("#298B8A", "#B4A7E1", "#E69F00"))

genome_wide_diversity_long %>% 
  ggplot(aes(x=fct_reorder(PopID, Tajimas.D), y=Tajimas.D, group = Treatment, fill=Treatment)) +
  geom_col(width=0.7, position=position_dodge(0.8)) + 
  theme_classic() + labs(x="Dispersal Tray", fill="Distance") +
  scale_fill_manual(values=c("#298B8A", "#B4A7E1", "#E69F00"))
```

### Summarize by Distance

```{r}
genome_wide_diversity_summary <- genome_wide_diversity_long %>% 
  group_by(Treatment) %>% 
  summarise(meanTheta.Pi=mean(Theta.Pi), 
            meanTheta.Watterson=mean(Theta.Watterson),
            meanTajimas.D=mean(Tajimas.D))
genome_wide_diversity_summary

genome_wide_diversity_summary %>% 
  ggplot(aes(x=Treatment, y=meanTheta.Pi, fill=Treatment)) +
  geom_col(width=0.7, position=position_dodge(0.8)) + 
  theme_classic() + #labs(x="Dispersal Tray", fill="Distance") +
  scale_y_continuous(expand = c(0, 0)) +
  scale_fill_manual(values=c("#298B8A", "#B4A7E1", "#E69F00"))

genome_wide_diversity_summary %>% 
  ggplot(aes(x=Treatment, y=meanTheta.Watterson, fill=Treatment)) +
  geom_col(width=0.7, position=position_dodge(0.8)) + 
  theme_classic() + #labs(x="Dispersal Tray", fill="Distance") +
  scale_y_continuous(expand = c(0, 0)) +
  scale_fill_manual(values=c("#298B8A", "#B4A7E1", "#E69F00"))

genome_wide_diversity_summary %>% 
  ggplot(aes(x=Treatment, y=meanTajimas.D, fill=Treatment)) +
  geom_col(width=0.7, position=position_dodge(0.8)) + 
  theme_classic() + #labs(x="Dispersal Tray", fill="Distance") +
  scale_fill_manual(values=c("#298B8A", "#B4A7E1", "#E69F00"))
```

## Genome-Wide FST

```{r}
genome_wide_FST <- read_csv("../../Analyzing Genetic Data/DispersalTraps/hapFire_Results/SNPs_to_analyze/fst_genome.csv") 
genome_wide_FST

genome_wide_FST_list <- read_csv("../../Analyzing Genetic Data/DispersalTraps/hapFire_Results/SNPs_to_analyze/fst-list_genome.csv")
head(genome_wide_FST_list)

genome_wide_FST_matrix <- read_csv("../../Analyzing Genetic Data/DispersalTraps/hapFire_Results/SNPs_to_analyze/fst-matrix_genome.csv")
genome_wide_FST_matrix

genome_wide_FST_comp_types <- genome_wide_FST_list %>% 
  select(Pop1=first, Pop2=second, fst) %>% 
  mutate(Pop1_Dist=if_else(str_detect(Pop1, "S"), "Source", #assign pop distances
                           if_else(Pop1=="D07" | Pop1=="D08" | Pop1=="D21" | Pop1=="D22", "Far", "Close")),
         Pop2_Dist=if_else(str_detect(Pop2,"S"), "Source",  #assign pop distances
                           if_else(Pop2=="D07" | Pop2=="D08" | Pop2=="D21" | Pop2=="D22", "Far", "Close")),
         Comparison_Type=if_else(Pop1_Dist=="Close" & Pop2_Dist=="Close", "Close-Close", #assign comparison types 
                         if_else(Pop1_Dist=="Far" & Pop2_Dist=="Far", "Far-Far", 
                         if_else(Pop1_Dist=="Source" & Pop2_Dist=="Source", "Source-Source",
                         if_else(Pop1_Dist=="Source" & Pop2_Dist=="Close", "Source-Close",    
                         if_else(Pop1_Dist=="Source" & Pop2_Dist=="Far", "Source-Far",
                         if_else(Pop1_Dist=="Close" & Pop2_Dist=="Source", "Source-Close",
                         if_else(Pop1_Dist=="Far" & Pop2_Dist=="Source", "Source-Far", "Close-Far"))))))))

genome_wide_FST_comp_types %>% 
  ggplot(aes(x=Comparison_Type, y=fst)) +
  geom_boxplot()

genome_wide_FST_SAMEBLOCKS <- genome_wide_FST_comp_types %>% 
  left_join(sample_info, by=join_by(Pop1==PopID)) %>% #add in block info 
  select(Pop1:Comparison_Type, Pop1_Block=Block) %>% 
  left_join(sample_info, by=join_by(Pop2==PopID)) %>% #add in block info  
  select(Pop1:Comparison_Type, Pop1_Block, Pop2_Block=Block) %>% 
  mutate(Same_Block=if_else(Pop1_Block==Pop2_Block, TRUE, FALSE)) %>% 
  filter(Same_Block==TRUE) #only keep comparisons within the same block 

genome_wide_FST_SAMEBLOCKS %>% 
  ggplot(aes(x=Comparison_Type, y=fst)) +
  geom_boxplot()

genome_wide_FST_summary <- genome_wide_FST_SAMEBLOCKS %>% 
  group_by(Comparison_Type, Pop1_Block) %>% 
  summarise(meanFst=mean(fst), semFst=sem(fst))
genome_wide_FST_summary
genome_wide_FST_summary %>% 
  ggplot(aes(x=Comparison_Type, y=meanFst)) +
  geom_col(width = 0.7, position = position_dodge(0.8)) + 
  geom_errorbar(aes(ymin=meanFst-semFst, ymax=meanFst+semFst), width=.2, position = position_dodge(0.8)) + 
  labs(x="Comparison Type", y="Avg Genome-Wide FST") +
  scale_y_continuous(expand = c(0, 0)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  facet_wrap(~Pop1_Block)


```
